### DISCUSSION TITLE
Why interface{} value is considered as time.Time in binary minus operation

### DISCUSSION THREAD

**Discussion by IgorKalet** [antonmedv/expr#401] [General]
Compilation of quite simple expression raises the following error:
```
invalid operation: / (mismatched types time.Time and interface {}) (1:13)
 | (a - b + c) / d
 | ............^
```
It happens because the binary minus processing has the following check:
https://github.com/antonmedv/expr/blob/2eb6a03aacf9ca8a87685a6f3b1adc3e8473b295/checker/checker.go#L262
And function isTime ends with the following:
https://github.com/antonmedv/expr/blob/2eb6a03aacf9ca8a87685a6f3b1adc3e8473b295/checker/types.go#L101

Changing of `return isAny(t)` to `return false` solves this problem.

BTW Compilation of ` (a + -b + c) / d` and ` (a + c - b) / d` expressions are ok w/o above change.

**Comment by antonmedv** [comment]
Please, provide full example.

**Reply by IgorKalet** [reply]
Here you are
[main.txt](https://github.com/antonmedv/expr/files/12330502/main.txt)

**Comment by antonmedv** [comment]
Yes, looks like a problem with isTime() function. 

Fixed and added regression test in 23efe7b36c3e627d35e6afaec0d6f45f1ed69601

Thanks for reporting.
